api/v1alpha1/nicclusterpolicy_webhook.go

```diff
@@ -0,0 +1,304 @@
+			"Invalid json configuration of rdmaSharedDevicePluginConfig"+err.Error()))
+	} else if result.Valid() {
+		configListInterface := rdmaSharedDevicePluginConfigJSON["configList"]
+		configList, _ := configListInterface.([]interface{})
+		for _, configInterface := range configList {
+			config := configInterface.(map[string]interface{})
+			resourceName := config["resourceName"].(string)
+			if !isValidRdmaSharedDevicePluginResourceName(resourceName) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"),
+					dp.Config, "Invalid Resource name, it must consist of alphanumeric characters, "+
+						"'-', '_' or '.', and must start and end with an alphanumeric character "+
+						"(e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0"+
+						"-9_.]*)?[A-Za-z0-9]')"))
+			}
+		}
+	} else {
+		for _, ResultErr := range result.Errors() {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("Config"), dp.Config, ResultErr.Description()))
+		}
+	}
+	return allErrs
+}
+
+// validate is a helper function to perform validation for IBKubernetesSpec.
+func (ibk *IBKubernetesSpec) validate(fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeStart) || !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeEnd) {
+		if !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeStart) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("pKeyGUIDPoolRangeStart"),
+				ibk.PKeyGUIDPoolRangeStart, "pKeyGUIDPoolRangeStart must be a valid GUID format:"+
+					"xx:xx:xx:xx:xx:xx:xx:xx with Hexa numbers"))
+		}
+		if !isValidPKeyGUID(ibk.PKeyGUIDPoolRangeEnd) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("pKeyGUIDPoolRangeEnd"),
+				ibk.PKeyGUIDPoolRangeEnd, "pKeyGUIDPoolRangeEnd must be a valid GUID format: "+
+					"xx:xx:xx:xx:xx:xx:xx:xx with Hexa numbers"))
+		}
+		return allErrs
+	} else if !isValidPKeyRange(ibk.PKeyGUIDPoolRangeStart, ibk.PKeyGUIDPoolRangeEnd) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("pKeyGUIDPoolRangeEnd"),
+			ibk.PKeyGUIDPoolRangeEnd, "pKeyGUIDPoolRangeEnd must not be greater than 02:FF:FF:FF:FF:FF:FF:FF"))
+	}
+	return allErrs
+}
+
+// isValidPKeyGUID checks if a given string is a valid GUID format.
+func isValidPKeyGUID(guid string) bool {
+	PKeyGUIDPattern := `^([0-9A-Fa-f]{2}:){7}([0-9A-Fa-f]{2})$`
+	PKeyGUIDRegex := regexp.MustCompile(PKeyGUIDPattern)
+	return PKeyGUIDRegex.MatchString(guid)
+}
+
+// isValidPKeyRange checks if range of startGUID and endGUID sis valid
+func isValidPKeyRange(startGUID, endGUID string) bool {
+	startGUIDWithoutSeparator := strings.ReplaceAll(startGUID, ":", "")
+	endGUIDWithoutSeparator := strings.ReplaceAll(endGUID, ":", "")
+
+	startGUIDIntValue := new(big.Int)
+	endGUIDIntValue := new(big.Int)
+	startGUIDIntValue, _ = startGUIDIntValue.SetString(startGUIDWithoutSeparator, 16)
+	endGUIDIntValue, _ = endGUIDIntValue.SetString(endGUIDWithoutSeparator, 16)
```

Comment:
`SetString` also has a second `bool` parameter indicating the success of the operation. I'd say we use it too. For example, if GUID value is more than 02:FF:FF:FF:FF:FF:FF:FF

Reply:
The only validation here that the range is OK and in a valid format.